container-engine-lib/lib/backend_impls/docker/docker_kurtosis_backend/logs_aggregator_functions/shared_helpers.go

```diff
@@ -0,0 +1,106 @@
+package logs_aggregator_functions
+
+import (
+	"context"
+	"github.com/kurtosis-tech/kurtosis/container-engine-lib/lib/backend_impls/docker/docker_kurtosis_backend/consts"
+	"github.com/kurtosis-tech/kurtosis/container-engine-lib/lib/backend_impls/docker/docker_manager"
+	"github.com/kurtosis-tech/kurtosis/container-engine-lib/lib/backend_impls/docker/docker_manager/types"
+	"github.com/kurtosis-tech/kurtosis/container-engine-lib/lib/backend_impls/docker/object_attributes_provider/label_key_consts"
+	"github.com/kurtosis-tech/kurtosis/container-engine-lib/lib/backend_impls/docker/object_attributes_provider/label_value_consts"
+	"github.com/kurtosis-tech/kurtosis/container-engine-lib/lib/backend_interface/objects/container_status"
+	"github.com/kurtosis-tech/kurtosis/container-engine-lib/lib/backend_interface/objects/logs_aggregator"
+	"github.com/kurtosis-tech/stacktrace"
+	"net"
+)
+
+const (
+	shouldShowStoppedLogsAggregatorContainers = true
+)
+
+// Returns nil [LogsAggregator] object if no container is found
+func getLogsAggregatorObjectAndContainerId(
+	ctx context.Context,
+	dockerManager *docker_manager.DockerManager,
+) (*logs_aggregator.LogsAggregator, string, error) {
+	logsAggregatorContainer, err := getLogsAggregatorContainer(ctx, dockerManager)
+	if err != nil {
+		return nil, "", stacktrace.Propagate(err, "An error occurred getting all logs aggregator containers")
+	}
+	if logsAggregatorContainer == nil {
+		return nil, "", nil
+	}
+
+	logsAggregatorContainerID := logsAggregatorContainer.GetId()
+
+	logsAggregatorObject, err := getLogsAggregatorObjectFromContainerInfo(
+		ctx,
+		logsAggregatorContainerID,
+		logsAggregatorContainer.GetStatus(),
+		dockerManager,
+	)
+	if err != nil {
+		return nil, "", stacktrace.Propagate(err, "An error occurred getting the logs Aggregator object using container ID '%v', labels '%+v' and the status '%v'", logsAggregatorContainer.GetId(), logsAggregatorContainer.GetLabels(), logsAggregatorContainer.GetStatus())
+	}
+
+	return logsAggregatorObject, logsAggregatorContainerID, nil
+}
+
+// Returns nil [Container] object if no logs aggregator container is found
+func getLogsAggregatorContainer(ctx context.Context, dockerManager *docker_manager.DockerManager) (*types.Container, error) {
```

Comment:
nit: when you need to represent the fact that no object were found (and this is an acceptable result for this function), I like to return a boolean. Here it would be `*types.Container, bool, error`. 
I agree it can seem a unnecessary, but it makes things a lot more explicit, via the function signature, that both the error and the returned object can be nil, and then it forces the consumer to handle this case.

Without the boolean, the consumer not reading the comments on your function could easily think that when error is nil, the object is not nil, which would lead to an NPE downstream

Reply:
i like this pattern, will change!